### PR TITLE
Edit module command

### PR DESCRIPTION
**Pull request by cheeheng**
This pull request implements the following command:
- `edit-module INDEX ([m/MODULE_CODE] [t/MODULE_TITLE])`
- At least one of `MODULE_CODE` and `MODULE_TITLE` must be present

### CODE REVIEW

**Review by teoyuqi — COMMENTED** [review]
Just some minor nitpicks. LGTM otherwise 😃

File: src/main/java/seedu/address/logic/commands/EditModuleCommand.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        List<Module> lastShownList = model.getFilteredModuleList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
+        }
+
+        Module moduleToEdit = lastShownList.get(index.getZeroBased());
+        Module editedModule = createEditedModule(moduleToEdit, editModuleDescriptor);
+
+        if (!moduleToEdit.isSameModule(editedModule) && model.hasModule(editedModule)) {
```
[the remark this answers is not in the document]
Comment (cheeheng) [reply src/main/java/seedu/address/logic/commands/EditModuleCommand.java:67]: Yes, it does, thanks for pointing that out.

**Review by teoyuqi — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 2]
```diff
@@ -128,9 +129,41 @@ Example:
...
+* `m/`: To be followed by the new module code of the module to be edited on Plannit.
+* `t/`: To be followed by the new module title of the module to be edited on Plannit.
+
+Format: `edit-module INDEX ([m/MODULE_CODE] [t/MODULE_TITLE])`
+* `INDEX` is the currently displayed index number of the module you are editing for on the screen.
+* You cannot edit a module such that the new module code coincides with the module code of another 
+  module on Plannit (case-insensitive).
+* If `MODULE_TITLE` is empty, then the module title will be deleted from Plannit.
+
+Examples:
```
Comment (shwene) [inline docs/UserGuide.md:147]: Perhaps we could add the scenario where we only change the module title

**Review by shwene — DISMISSED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/EditModuleCommand.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        this.editModuleDescriptor = new EditModuleDescriptor(editModuleDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Module> lastShownList = model.getFilteredModuleList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
```
Comment (Tan-Jia-Rong) [inline src/main/java/seedu/address/logic/commands/EditModuleCommand.java:61]:
```suggestion
            throw new CommandException(MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
```

@AY2223S1-CS2103T-T10-1/developers Should we import via Messages.blah_blah or import the message as shown in suggested

**Review by Tan-Jia-Rong — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/EditModuleCommand.java (hunk 3 above)
Comment (cheeheng) [reply src/main/java/seedu/address/logic/commands/EditModuleCommand.java:62]: There are already `MESSAGE_SOMETHING` variables declared inside the file. Not explicitly referencing `Messages` class will likely cause confusion.

**Review by teoyuqi — DISMISSED** [review]
LGTM

**Review by shwene — APPROVED** [review]
LGTM

**Review by teoyuqi — APPROVED** [review]
LGTM

**Review by Tan-Jia-Rong — APPROVED** [review]
LGTM

**Pull request merged**